### PR TITLE
Replace `lazy_static` with `std::sync::LazyLock`

### LINKED ISSUE #15860: remove lazy_static and use std::sync::LazyLock
### Check for existing issues

- [X] Completed

### Describe the feature

rust 1.80 support LazyLock, lazy_static can be replaced

https://blog.rust-lang.org/2024/07/25/Rust-1.80.0.html

some reference: https://github.com/kkrt-labs/kakarot-rpc/pull/1315

### If applicable, add mockups / screenshots to help present your vision of the feature

_No response_

### PR DESCRIPTION
**Pull request by SinanGncgl**
Closes #15860 

Since rust std now supports LazyLock replacing lazy_static with it reduce the external dependency. 

Release Notes:

- N/A

### CODE REVIEW

**Comment by SinanGncgl** [comment]
> I think also once_cell crate can be replaced
> 
> for example [paradigmxyz/reth#9799](https://github.com/paradigmxyz/reth/issues/9799)

Yes, it can be a separate pr.

File: crates/sqlez_macros/src/sqlez_macros.rs [hunk 1]
```diff
@@ -1,12 +1,16 @@
 use proc_macro::{Delimiter, Span, TokenStream, TokenTree};
-use sqlez::thread_safe_connection::{locking_queue, ThreadSafeConnection};
 use syn::Error;
 
-lazy_static::lazy_static! {
-    static ref SQLITE: ThreadSafeConnection =  {
-        ThreadSafeConnection::new(":memory:", false, None, Some(locking_queue()))
-    };
-}
+#[cfg(not(target_os = "linux"))]
```
Comment (maxdeviant) [inline crates/sqlez_macros/src/sqlez_macros.rs:4]: Curious why we needed to add this target check here, since it wasn't there before?

Comment (SinanGncgl) [reply crates/sqlez_macros/src/sqlez_macros.rs:4]: I think the conditional check is necessary since we only use for SQLITE there for linux.

Comment (maxdeviant) [reply crates/sqlez_macros/src/sqlez_macros.rs:4]: But why wasn't it there before?

If it's not directly related to the `LazyLock` changes I would prefer it did not go in this PR.

Comment (maxdeviant) [reply crates/sqlez_macros/src/sqlez_macros.rs]: Alright, I figured out why we need this now:

`lazy_static!` automatically marks the generated static with `#[allow(dead_code)]`: https://github.com/rust-lang-nursery/lazy-static.rs/blob/cc81425f478f20eb286b3698ade99a99dd41a084/src/lib.rs#L150

**Review by maxdeviant — APPROVED** [review]
Thank you!

**Pull request merged**